src/grid_interpolator.cc

```diff
@@ -0,0 +1,104 @@
+#include "flyft/grid_interpolator.h"
+
+#include <fstream> 
+#include <cassert>
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator(std::string s)
+    :GridInterpolator(s, 0, 0, 0)
+    {
+    // Default constructor for the ThreeDimensionIndex
+    }
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk):
+    n_(ni, nj ,nk)
+    {
+    double dx, dy, dz;
+    std::ifstream indata;
+    indata.open(s);
+    if(!indata.is_open())
+        {
+        std::cout<<"File failed to open"<<std::endl;
+        }
+    indata>>ni>>nj>>nk>>dx>>dy>>dz;
+    indata.close();
+    GridInterpolator(s, ni, nj, nk, dx, dy, dz);
+    }    
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk, double dx, double dy, double dz):
+    n_(ni, nj, nk), dx_(dx), dy_(dy), dz_(dz)
+    {
+    data_ = new double[n_.size()];  
+    std::ifstream indata;
+    indata.open(s);
+    while(indata.good()) 
+        {
+        double value;
+        // Ignore the first line of the input file
+        indata.ignore(6,'\n');
+        while(indata >> value)
+            {
+            for (int i = 0; i < ni; i++)
+                {
+                for (int j = 0; j < nj; j++)
+                    {
+                    for (int k = 0; k < nk; k++)
+                        {
+                        data_[n_(i, j, k)] = value;
+                        }
+                    }
+                }
+            }
+        }
+    // Close the file.
+    indata.close();
+    }
+
+double GridInterpolator::operator()(double x, double y, double z) const
+    {
+    
+    int bin_x = std::floor(x/dx_); 
+    int bin_y = std::floor(y/dy_); 
+    int bin_z = std::floor(z/dz_); 
+    int ni, nj , nk;
+    std::tie(ni, nj, nk) = n_.getBounds();
+    assert(bin_x >= 0 || bin_x < ni);
+    assert(bin_y >= 0 || bin_y < nj);
+    assert(bin_z >= 0 || bin_z < nk);
```

Comment:
both of these need to be true at the same time:
```suggestion
    assert(bin_x >= 0 && bin_x < ni);
    assert(bin_y >= 0 && bin_y < nj);
    assert(bin_z >= 0 && bin_z < nk);
```